### DISCUSSION TITLE
Cancel request

### DISCUSSION THREAD

**Discussion by flrnt** [DamianOsipiuk/vue-query#140] [Q&A]
Is there a way to cancel a request, via the native Web API [AbortController](https://developer.mozilla.org/en-US/docs/Web/API/AbortController) ?

**Comment by DamianOsipiuk** [comment]
Hi, please refer to this page: https://react-query.tanstack.com/guides/query-cancellation

**Comment by flrnt** [comment]
Ok, I see. I quickly wrote something related to the functionality I can use on my project. I will test. Really nice stuff btw 

```js
// composables/useCancelableQuery.js

export function useCancelableQuery = (key, url, params = {}) => {
  const controller = new AbortController()
  const { signal } = controller

  const { isLoading, isError, data, error } = useQuery(key, () => axios.get(url, { ...params, signal }))
  const doCancel = () => { controller.abort() }

  return {
    isLoading,
    isError,
    data,
    error,
    doCancel
  }
}
```
